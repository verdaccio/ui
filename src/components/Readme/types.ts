export interface IProps {
  description: string;
}
