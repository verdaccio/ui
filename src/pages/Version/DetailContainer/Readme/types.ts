export interface Props {
  description: string;
}
