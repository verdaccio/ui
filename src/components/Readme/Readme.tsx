import React from 'react';
import 'github-markdown-css';

import { Props } from './types';

const Readme: React.FC<Props> = ({ description }) => <div className="markdown-body" dangerouslySetInnerHTML={{ __html: description }} />;

export default Readme;
