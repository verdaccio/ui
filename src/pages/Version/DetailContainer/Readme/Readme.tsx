import styled from '@emotion/styled';
import React from 'react';
import 'github-markdown-css';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import { Props } from './types';

const Readme: React.FC<Props> = ({ description }) => (
  <Wrapper className="markdown-body" dangerouslySetInnerHTML={{ __html: description }} />
);

export default Readme;

const Wrapper = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  background: theme?.palette.white,
  color: theme?.palette.black,
  padding: theme?.spacing(2, 3),
  ul: {
    listStyle: 'disc',
  },
}));
