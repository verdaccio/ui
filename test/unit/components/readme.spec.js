/**
 * Readme component
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import Readme from '../../../src/webui/components/Readme/index';

describe('<Readme /> component', () => {
  test('should load the component in default state', () => {
    const wrapper = mount(<Readme description={"test"} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should dangerously set html', () => {
    const wrapper = shallow(<Readme description={"<h1>This is a test string</h1>"} />);
    expect(wrapper.html()).toEqual(
      '<div class="markdown-body"><h1>This is a test string</h1></div>'
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
