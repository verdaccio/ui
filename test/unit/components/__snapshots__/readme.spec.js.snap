// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Readme /> component should dangerously set html 1`] = `"<div class=\\"markdown-body\\"><h1>This is a test string</h1></div>"`;

exports[`<Readme /> component should load the component in default state 1`] = `"<div class=\\"markdown-body\\">test</div>"`;
